src/extensions/step_handlers/wim_handler/src/format_drive.cpp

```diff
@@ -17,8 +18,36 @@
         }                    \
     }
 
-HRESULT FormatDrive(char driveLetter)
+static HRESULT DoesVolumeLabelExist(char driveLetter, const std::string& volumeLabel)
 {
+    char rootPathName[] = "X:\\";
+    rootPathName[0] = driveLetter;
+
+    LPSTR volumeName = const_cast<LPSTR>(volumeLabel.c_str());
+    DWORD nVolumeNameSize;
+    CHR(SizeTToDWord(volumeLabel.length(), &nVolumeNameSize));
+    CHR(DWordAdd(nVolumeNameSize, 1, &nVolumeNameSize));
+
+    if(!GetVolumeInformationA(
+        rootPathName,
+        volumeName,
+        nVolumeNameSize,
+        nullptr, // lpVolumeSerialNumber
+        nullptr, // lpMaxComponentLen
+        nullptr, // lpFileSystemFlags
+        nullptr, // lpFileSystemNameBuffer
+        0)) // nFileSystemNameSize
+    {
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
+
+    return S_OK;
+}
+
+HRESULT FormatDrive(char driveLetter, const std::string& volumeLabel)
+{
+    CHR(DoesVolumeLabelExist(driveLetter, volumeLabel));
```

Comment:
this doesn't belong here.  it belongs in the caller, as if I call FormatDrive I expect it to format the drive.